### PR TITLE
Add the option to copy a list of files to ARTIFACTORY_HOME on startup

### LINKED ISSUE #181: binarystore.xml is not updateable after initial helm install of artifactory-ha
**Is this a BUG REPORT or FEATURE REQUEST?**: BUG

**Version of Helm and Kubernetes**: Helm v2.11.0, Kubernetes v1.10.11

**Which chart**: artifactory-ha:8.7.0

**What happened**: binarystore.xml is not updateable after initial ```helm install``` command

**What you expected to happen**: binarystore.xml is propagated from /artifactory_extra_conf to /var/opt/jfrog/artifactory/etc on ```helm upgrade``` if it's changed

binarystore.xml is not updateable after initial ```helm install```

I see that binarystore.xml is mounted from configMap in the chart and Docker image ```docker.bintray.io/jfrog/artifactory-pro``` has an entrypoint-artifactory.sh which should propagate the file from a mentioned folder, but it never happen:
```
addExtraConfFiles () {
    logger "Adding extra configuration files to ${ARTIFACTORY_HOME}/etc if any exist"

    # If directory not empty
    if [ -d "${ARTIFACTORY_EXTRA_CONF}" ] && [ "$(ls -A ${ARTIFACTORY_EXTRA_CONF})" ]; then
        logger "Adding files from ${ARTIFACTORY_EXTRA_CONF} to ${ARTIFACTORY_HOME}/etc"
        cp -rfv ${ARTIFACTORY_EXTRA_CONF}/* ${ARTIFACTORY_HOME}/etc || errorExit "Copy files from ${ARTIFACTORY_EXTRA_CONF} to ${ARTIFACTORY_HOME}/etc failed"
    fi
}
```
Obviously it's happen becauase of next test in entrypoint-artifactory.sh
```
# In case data dirs are missing or not mounted, need to create them
setupDataDirs () {
    logger "Setting up Artifactory data directories if missing"
    if [ ! -d ${ARTIFACTORY_DATA}/etc ]; then
        mkdir -pv ${ARTIFACTORY_DATA}/etc || errorExit "Failed creating ${ARTIFACTORY_DATA}/etc"

        # Add extra conf files to a newly created etc/ only!
        addExtraConfFiles
    fi
    [ -d ${ARTIFACTORY_DATA}/data ]        || mkdir -pv ${ARTIFACTORY_DATA}/data        || errorExit "Creating ${ARTIFACTORY_DATA}/data failed"
    [ -d ${ARTIFACTORY_DATA}/logs ]        || mkdir -pv ${ARTIFACTORY_DATA}/logs        || errorExit "Creating ${ARTIFACTORY_DATA}/logs failed"
    [ -d ${ARTIFACTORY_DATA}/backup ]      || mkdir -pv ${ARTIFACTORY_DATA}/backup      || errorExit "Creating ${ARTIFACTORY_DATA}/backup failed"
    [ -d ${ARTIFACTORY_DATA}/access ]      || mkdir -pv ${ARTIFACTORY_DATA}/access      || errorExit "Creating ${ARTIFACTORY_DATA}/access failed"
    [ -d ${ARTIFACTORY_HOME}/run ]         || mkdir -pv ${ARTIFACTORY_HOME}/run         || errorExit "Creating ${ARTIFACTORY_HOME}/run failed"
    [ -d ${ARTIFACTORY_HOME}/etc/plugins ] || mkdir -pv ${ARTIFACTORY_HOME}/etc/plugins || errorExit "Creating ${ARTIFACTORY_HOME}/etc/plugins failed"
    [ -d ${REPLICATOR_DATA} ]              || mkdir -pv ${REPLICATOR_DATA}              || errorExit "Creating ${REPLICATOR_DATA} failed"
}
```
After initial ```helm install``` persistent volume is created and populated with ${ARTIFACTORY_DATA}/etc so the bash test above will never happen on subsequent ```helm update```s

**How to reproduce it**:
Steps to reproduce:
1. Initially deploy artifactory-ha with ```helm install``` and with operator provided helm values, e.g. -f values.yaml
2. Update binarystore.xml related values in values.yaml, e.g. change IAM role
3. Update Artifactory workload using ```helm upgrade``` with operator provided values, e.g. -f values.yaml and ```--recreate-pods``` parameter
4. ```diff /artifactory_extra_conf/binarystore.xml /var/opt/jfrog/artifactory/etc/binarystore.xml``` - files will be different and binarystore.xml isn't propagated from /artifactory_extra_conf to /var/opt/jfrog/artifactory/etc

### PR DESCRIPTION
**Pull request by danielezer**
#### PR Checklist
[Place an '[x]' (no spaces) in all applicable fields. Please remove unrelated fields.]
- [x] Chart Version bumped
- [x] CHANGELOG.md updated
- [x] Variables and other changes are documented in the README.md

**What this PR does / why we need it**:
Adds the option to copy a list of files on startup to ARTIFSCTORY_HOME

**Which issue this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close that issue when PR gets merged)*: fixes #
fixes #181

### CODE REVIEW

**Comment by danielezer** [comment]
@eldada do you think we need to copy the files in the member nodes in HA? or is the primary enough?

File: stable/artifactory/Chart.yaml [hunk 1]
```diff
@@ -1,7 +1,7 @@
 apiVersion: v1
 name: artifactory
 home: https://www.jfrog.com/artifactory/
-version: 7.13.0
+version: 7.12.9
```
Comment (jainishshah17) [inline stable/artifactory/Chart.yaml]: @danielezer chart version should be 7.13.1.

Comment (danielezer) [reply stable/artifactory/Chart.yaml]: @jainishshah17 changed

File: stable/artifactory/CHANGELOG.md [hunk 2]
```diff
@@ -1,7 +1,7 @@
 # JFrog Artifactory Chart Changelog
 All changes to this chart will be documented in this file.
 
-## [7.12.19] - Mar 27, 2019
+## [7.13.1] - Mar 27, 2019
```
Comment (jainishshah17) [inline stable/artifactory/CHANGELOG.md]: @danielezer Can you please fix order of versions?

Comment (danielezer) [reply stable/artifactory/CHANGELOG.md]: @jainishshah17 Done

File: stable/artifactory-ha/templates/artifactory-node-statefulset.yaml [hunk 3]
```diff
@@ -80,7 +80,6 @@ spec:
         imagePullPolicy: {{ .Values.artifactory.image.pullPolicy }}
         securityContext:
           allowPrivilegeEscalation: false
-      {{- if .Values.artifactory.preStartCommand }}
```
Comment (jainishshah17) [inline stable/artifactory-ha/templates/artifactory-node-statefulset.yaml]: @danielezer Can you please add this line.

**Comment by jainishshah17** [comment]
@danielezer I thinks it's good idea to add this feature in both primary and member nodes

File: stable/artifactory-ha/templates/artifactory-node-statefulset.yaml (hunk 3 above)
Comment (danielezer) [reply stable/artifactory-ha/templates/artifactory-node-statefulset.yaml]: Nice catch. sorry about that. Thanks

**Comment by danielezer** [comment]
> @danielezer I thinks it's good idea to add this feature in both primary and member nodes

@jainishshah17 I asked @eldada and he said adding it to the primary node only should be enough

**Comment by danielezer** [comment]
@eldada can you please review?

**Pull request merged**